core/include/cubos/core/al/audio_device.hpp

```diff
@@ -0,0 +1,152 @@
+#ifndef CUBOS_CORE_AL_AUDIO_DEVICE_HPP
+#define CUBOS_CORE_AL_AUDIO_DEVICE_HPP
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <glm/glm.hpp>
+
+namespace cubos::core::al
+{
+    namespace impl
+    {
+        class Buffer;
+        class Source;
+    } // namespace impl
+
+    using Buffer = std::shared_ptr<impl::Buffer>;
+    using Source = std::shared_ptr<impl::Source>;
+
+    /// Possible audio formats.
+    enum class Format
+    {
+        Mono8,
+        Mono16,
+        Stereo8,
+        Stereo16,
+    };
+
+    /// Abstract audio device type, used to wrap low-level audio rendering APIs such as OpenAL.
+    /// @see al::OALRenderDevice.
+    class AudioDevice
+    {
+    public:
+        AudioDevice() = default;
+        virtual ~AudioDevice() = default;
+        AudioDevice(const AudioDevice&) = delete;
+
+        /// Creates an audio device from a given device specifier.
+        /// @param specifier The device specifier (empty for default).
+        /// @returns The audio device.
+        static std::shared_ptr<AudioDevice> create(std::string specifier = "");
+
+        /// Enumerates the available devices.
+        /// @param devices The vector to fill with the available devices.
+        static void enumerateDevices(std::vector<std::string>& devices);
```

Comment:
It's better to let the implementation create the array, instead of asking users to create the object before hand:
```suggestion
        static std::vector<std::string> enumerateDevices();
```